test/setup.sh

```diff
@@ -11,27 +12,52 @@ cd "$(dirname "${BASH_SOURCE[0]}")"
 echo "Copying weave images, scripts, and certificates to hosts, and"
 echo "  prefetch test images"
 
+exists_on() {
+    docker_on $1 inspect --format=" " $2 >/dev/null 2>&1
+}
+
+pull_if_missing() {
+    if ! exists_on "$1" "$2" ; then
+        echo "Pulling $2 on $1..."
+        docker_on $1 pull $2 >/dev/null 2>&1
```

Comment:
I don't think this should redirect stderr; if something goes wrong with the pull we wan't to know about it.